ssh/tailssh/tailssh_test.go

```diff
@@ -155,3 +168,75 @@ func TestMatchRule(t *testing.T) {
 }
 
 func timePtr(t time.Time) *time.Time { return &t }
+
+func TestSSH(t *testing.T) {
+	ml := new(tstest.MemLogger)
+	var logf logger.Logf = ml.Logf
+	eng, err := wgengine.NewFakeUserspaceEngine(logf, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	lb, err := ipnlocal.NewLocalBackend(logf, "",
+		new(ipn.MemoryStore),
+		new(tsdial.Dialer),
+		eng, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer lb.Shutdown()
+	dir := t.TempDir()
+	lb.SetVarRoot(dir)
+
+	srv := &server{lb, logf}
+	ss, err := srv.newSSHServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	u, err := user.Current()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ci := &sshConnInfo{
+		sshUser: "test",
+		srcIP:   netaddr.MustParseIP("1.2.3.4"),
+		node:    &tailcfg.Node{},
+		uprof:   &tailcfg.UserProfile{},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ss.Handler = func(s ssh.Session) {
+		srv.handleAcceptedSSH(ctx, s, ci, u)
+	}
+
+	ln, err := net.Listen("tcp4", "127.0.0.1:0")
```

Comment:
you can use https://pkg.go.dev/tailscale.com/net/nettest#Listener then you don't have to do the whole port dance.

Reply:
oh, oops. I thought you were calling from within the process. You're exec-ing out to `ssh`. This is fine.